arch/x86/core/CMakeLists.txt

```diff
@@ -10,6 +10,7 @@ endif()
 zephyr_compile_options_ifdef(CONFIG_COVERAGE_GCOV
```

Comment:
side note:

what's the difference between CONFIG_COVERAGE and CONFIG_COVERAGE_GCOV?

Reply:
`subsys/testsuite/Kconfig` said `Create Coverage data from hardware platform` for `COVERAGE_GCOV`.

Reply:
@andrewboie ```CONFIG_COVERAGE``` has existed for a long time, and has been used in the posix arch, to compile it with ```--coverage```, but in general to disable optimizations.
```CONFIG_COVERAGE_GCOV``` was added in #9717, it adds the extra gcov equivalent code to the embedded builds https://github.com/zephyrproject-rtos/zephyr/tree/master/subsys/testsuite/coverage to count the hits, and send out the report over UART once the test is done.

So being a bit more accurate I'd think this no-inline should be under a ```_ifdef(CONFIG_COVERAGE)``` instead